decoder.js

```diff
@@ -33,15 +43,94 @@ Decoder.prototype._writeOffset = function (chunk, offset, cb) {
     if (this._inBody) {
       offset = this._writeBody(chunk, offset, cb)
       if (offset === 0) return // chunk not consumed - _writeOffset will be called again when ready
-    } else {
+    } else if (this._inHead) {
       offset = this._writeHead(chunk, offset)
+    } else if (this._inInterleavedHead) {
+      offset = this._writeInterleavedHead(chunk, offset)
+    } else if (this._inInterleavedPacket) {
+      offset = this._writeInterleavedPacket(chunk, offset, cb)
+    } else {
+      if (chunk[offset] === INTERLEAVED_SIGN) {
+        offset = this._writeInterleavedHead(chunk, offset)
+      } else {
+        offset = this._writeHead(chunk, offset)
+      }
     }
   }
   cb()
 }
+Decoder.prototype._writeInterleavedHead = function (chunk, offset) {
+  if (this._interleavedHeaderOffset === 0) {
+    debug('start of interleaved header')
+    this._inInterleavedHead = true
+  }
+  if (chunk.length - offset < 4) {
+    chunk.copy(this._interleavedHeader, this._interleavedHeaderOffset, offset)
+    this._interleavedHeaderOffset += chunk.length - offset
+    return chunk.length
+  }
+
+  chunk.copy(this._interleavedHeader, this._interleavedHeaderOffset, offset, offset + INTERLEAVED_HEADER_BYTES)
+
+  debug('end of interleaved header')
+  this._inInterleavedHead = false
+
+  this._packet = new stream.PassThrough()
+  this._packet.channel = this._interleavedHeader.readInt8(1)
+  this._packet.size = this._interleavedHeader.readInt16BE(2)
+
+  this._interleavedHeaderOffset = 0
+  this._interleavedHeader = new Buffer(INTERLEAVED_HEADER_BYTES)
```

Comment:
Probably you are right.

I just tried to use the same approach you used for RTSP Header to make it easier to understand.

As it is a fixed size reader header, maybe we should just remove line 83 and resuse the same buffer.